test/e2e/e2e_test.go

```diff
@@ -0,0 +1,279 @@
+
+	err = deployBroker(namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = registerBroker(namespace)
+	if err != nil {
+		fmt.Println(err)
+		t.Fatal(err)
+	}
+
+	return namespace
+}
+
+// cleanupTest will destroy a test namespace and all its resources.
+func cleanupTest(t *testing.T) {
+	namespace := namespaceForTest(t)
+
+	err := kubeClient.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+	if err != nil {
+		panic(err)
+	}
+}
+
+// namespaceForTest will return a namespace name based on the current test.
+func namespaceForTest(t *testing.T) string {
+	return fmt.Sprintf("aosb-e2e-%s", strings.ToLower(t.Name()))
+}
+
+// deployBroker will deploy the broker inside the specified namespace.
+func deployBroker(namespace string) error {
+	numOfReplicas := int32(1)
+
+	// Create deployment of the broker server.
+	_, err := kubeClient.AppsV1().Deployments(namespace).Create(&appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &numOfReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": name,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						v1.Container{
+							Name:            name,
+							Image:           image,
+							ImagePullPolicy: "Never",
+							Ports: []v1.ContainerPort{
+								v1.ContainerPort{
+									ContainerPort: int32(4000),
+								},
+							},
+							Env: []v1.EnvVar{
+								v1.EnvVar{
+									Name:  "ATLAS_GROUP_ID",
+									Value: atlasGroupID,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_BASE_URL",
+									Value: atlasBaseURL,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_PUBLIC_KEY",
+									Value: atlasPublicKey,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_PRIVATE_KEY",
+									Value: atlasPrivateKey,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_USERNAME",
+									Value: basicAuthUsername,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_PASSWORD",
+									Value: basicAuthPassword,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_HOST",
+									Value: "0.0.0.0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create service to expose broker deployment internally.
+	_, err = kubeClient.CoreV1().Services(namespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"app": name,
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"app": name,
+			},
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Protocol:   v1.ProtocolTCP,
+					Port:       80,
+					TargetPort: intstr.FromInt(4000),
+				},
+			},
+		},
+	})
+
+	return err
+}
+
+// registerBroker will register a previously deployed broker with the
```

Comment:
I read this to mean a broker that was deployed previously but no longer is. Do we actually mean register an existing/already deployed broker?

Reply:
I see you confusion. Referring to the broker deployed in `deployBroker` but "previously" is a bad choice of word. Updated now to read:
```
// registerBroker will register the broker deployed in deployBroker with the
// service catalog.
```